### PR TITLE
Letting schedules from the server override schedules from disk.

### PR DESCRIPTION
**Pull request by RonConescu**
The feature:  if we get a schedule from the server for an existing task, let the server schedule take control of that task.  We also prevent the disk copy of that schedule from retaking control of that task when the app next launches.

Also:
- Added a flowchart describing the business rules in the import method
- Reorganized the files related to the import process, both on disk and in the project

### CODE REVIEW

**Comment by RonConescu** [comment]
@YuanZhu-apple — Another one for you, sir!  —ron

File: APCAppCore/APCAppCore/Library/ScheduleImporter/APCScheduleImporter.m [hunk 1]
```diff
@@ -66,7 +66,7 @@
  Controls whether we compute and show (very) detailed debugging
  printouts.  This will always be NO in a release build.
  */
-static BOOL const kAPCShowDebugPrintouts = YES;
+static BOOL const kAPCShowDebugPrintouts = NO;
```
Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/ScheduleImporter/APCScheduleImporter.m:69]: Is this change intended?

File: APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m [hunk 2]
```diff
@@ -249,6 +264,23 @@ + (NSString *) stringFromDate: (NSDate *) date
     return result;
 }
 
+- (NSString *) stringWithMillisecondsFromDate: (NSDate *) date
```
Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: Is this method necessary?

**Comment by YuanZhu-apple** [comment]
@RonConescu My review is done.

File: APCAppCore/APCAppCore/Library/ScheduleImporter/APCScheduleImporter.m (hunk 1 above)
Comment (RonConescu) [inline APCAppCore/APCAppCore/Library/ScheduleImporter/APCScheduleImporter.m:69]: Yes.  The printouts are very verbose.  I wanted them to be available to anyone who needs them, but not get in the way unless asked for.

Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/ScheduleImporter/APCScheduleImporter.m:69]: Sure.

File: APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m (hunk 2 above)
Comment (RonConescu) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: Yes.  During the above printouts, I use this millisecond timestamp in the header so I can uniquely identify the printouts, and search for them reliably.  I put it in the debugPrinter class to make it easier to use the same date-formatting as other printouts emitted by the debugPrinter.

Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: My question was why not use class method directly? It seems this instance method is not required.

Comment (RonConescu) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: Ah.  Because most of the time, when I use it, I already have a debugPrinter object allocated, so I've found it to be less code, and more readable, to simply ask the printer object to format something for me.

Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: But `APCScheduleDebugPrinter` class is always available as well.

Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: Perhaps change to `[printer.class stringWithMillisecondsFromDate:date]`?

Comment (RonConescu) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: I'll certainly change it if you feel strongly about it.  My thoughts:  The debugPrinter class also has a `-stringFromDate:` method, which prints a lower-resolution time stamp.  I've used `-stringFromDate:` in a bunch of places, inside and outside the class.  In general, when I'm writing, I navigate the code using Xcode's context menu, and I'll name classes and methods so they alphabetize together in the pop-up menu.  So, here, when I start to type the name of the method, like `[printer str...`, I've found it helpful to have that pop-up remind me I've got 2 versions of the method -- one high-res and one low-res.

Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: Another thought: add `stringWithMilliseconds` to a NSDate Category? So no need to involve printer in this case.

Comment (RonConescu) [inline APCAppCore/APCAppCore/Library/Scheduler/APCScheduleDebugPrinter.m]: In this case, the method is tied to what I'm doing with the debugPrinter.  It's not intended to be a general-purpose method.  It's intended as a way of making sure I get consistent date formatting when I'm analyzing tasks and schedules, so we can compare dates across schedule printouts and know we're seeing dates in the same format.  It also lets me normalize the dates when I print a bunch of schedules in a columnar format.

**Comment by RonConescu** [comment]
@YuanZhu-apple — I've changed it to implement your suggestion:  [printer.class stringWithMillisecondsFromDate...].  That's a very clean idea; thank you.

**Comment by YuanZhu-apple** [comment]
@RonConescu Thanks!
